# IS 362 – Week 5 Assignment

Alyssa Ellis

This assignment uses Python and pandas to answer the following questions:

1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12, 2013, which New York area airport had the windiest weather?

In [1]:
import pandas as pd

In [3]:
airports = pd.read_csv("/Users/alyssaellis/Desktop/airports.csv")
weather = pd.read_csv("/Users/alyssaellis/Desktop/weather.csv")

In [4]:
airports.head()

,faa,name,lat,lon,alt,tz,dst,tzone
0,04G,Lansdowne Airport,41.130472,-80.619583,1044,-5,A,America/New_York
1,06A,Moton Field Municipal Airport,32.460572,-85.680028,264,-6,A,America/Chicago
2,06C,Schaumburg Regional,41.989341,-88.101243,801,-6,A,America/Chicago
3,06N,Randall Airport,41.431912,-74.391561,523,-5,A,America/New_York
4,09J,Jekyll Island Airport,31.074472,-81.427778,11,-5,A,America/New_York


## Question 1: What is the northernmost airport in the United States?

I first sorted the airport data by latitude from highest to lowest. 
Since higher latitude values represent locations farther north, this 
shows the northernmost airports in the dataset.

In [5]:
northern_airports = airports.sort_values("lat", ascending=False)

northern_airports[["faa", "name", "lat", "lon"]].head(5)

,faa,name,lat,lon
417,EEN,Dillant Hopkins Airport,72.270833,42.898333
230,BRW,Wiley Post Will Rogers Mem,71.285446,-156.766003
110,AIN,Wainwright Airport,70.638056,-159.994722
708,K03,Wainwright As,70.613378,-159.860350
152,ATK,Atqasuk Edward Burnell Sr Memorial Airport,70.467300,-157.436000


### Answer

At first, the data showed Dillant-Hopkins Airport (EEN) as the northernmost airport. However, the coordinates in the dataset are incorrect because this airport is actually located in New Hampshire. After looking at the next results and checking the location, I found that **Wiley Post-Will Rogers Memorial Airport (BRW) in Alaska** is the northernmost airport, at about 71.29° N.

## Question 2: What is the easternmost airport in the United States?

I examined the longitude values to find the easternmost airports. 
Because longitude wraps around at the International Date Line, I 
looked at airports at both extremes of the longitude data before 
determining the answer.

In [6]:
eastern_airports = airports.sort_values("lon", ascending=False)

eastern_airports[["faa", "name", "lat", "lon"]].head(5)

,faa,name,lat,lon
1290,SYA,Eareckson As,52.712275,174.113620
942,MYF,Montgomery Field,32.475900,117.759000
396,DVT,Deer Valley Municipal Airport,33.411700,112.457000
417,EEN,Dillant Hopkins Airport,72.270833,42.898333
444,EPM,Eastport Municipal Airport,44.910111,-67.012694


In [7]:
western_longitudes = airports.sort_values("lon", ascending=True)

western_longitudes[["faa", "name", "lat", "lon"]].head(5)

,faa,name,lat,lon
95,ADK,Adak Airport,51.878000,-176.646000
112,AKB,Atka Airport,52.220278,-174.206389
522,GAM,Gambell Airport,63.766800,-171.733000
1281,SVA,Savoonga Airport,63.686400,-170.493000
1251,SNP,St Paul Island,57.167333,-170.220444


### Answer

The longitude data had some incorrect coordinates, so I had to look at the results more closely instead of just choosing the first one. Eareckson Air Station (SYA) is located on Shemya Island, Alaska, and its longitude of about 174.11° E is valid. Because Alaska crosses into the Eastern Hemisphere, **Eareckson Air Station (SYA)** is the easternmost airport in the dataset.

## Question 3: On February 12, 2013, which New York area airport had the windiest weather?

I filtered the weather data to include only observations from 
February 12, 2013. I then compared the wind speed observations for 
the three New York area airports.

In [8]:
feb12_weather = weather[
    (weather["year"] == 2013) &
    (weather["month"] == 2) &
    (weather["day"] == 12)
]

feb12_weather.head()

,origin,year,month,day,hour,temp,dewp,humid,wind_dir,wind_speed,wind_gust,precip,pressure,visib,time_hour
1006,EWR,2013,2,12,0,39.92,39.02,96.55,240.0,6.90468,NaN,0.0,1006.9,10.0,2013-02-12T05:00:00Z
1007,EWR,2013,2,12,1,39.92,37.94,92.56,250.0,9.20624,NaN,0.0,1007.2,10.0,2013-02-12T06:00:00Z
1008,EWR,2013,2,12,2,39.92,28.04,62.21,270.0,20.71404,25.31716,0.0,1007.8,10.0,2013-02-12T07:00:00Z
1009,EWR,2013,2,12,3,39.02,26.96,61.63,260.0,1048.36058,NaN,0.0,1008.3,10.0,2013-02-12T08:00:00Z
1010,EWR,2013,2,12,4,39.02,26.96,64.29,280.0,12.65858,NaN,0.0,1008.8,10.0,2013-02-12T09:00:00Z


In [9]:
feb12_weather[["origin", "hour", "wind_speed"]].sort_values(
    "wind_speed", ascending=False
).head(15)

,origin,hour,wind_speed
1009,EWR,3,1048.36058
18417,LGA,2,23.01560
1018,EWR,12,21.86482
18428,LGA,13,21.86482
18429,LGA,14,20.71404
1008,EWR,2,20.71404
9717,JFK,8,20.71404
9716,JFK,7,19.56326
18427,LGA,12,19.56326
18416,LGA,1,19.56326


In [10]:
feb12_weather.groupby("origin")["wind_speed"].describe()

,count,mean,std,min,25%,50%,75%,max
origin,,,,,,,,
EWR,24.0,56.38822,211.336876,4.60312,10.93241,13.80936,17.261700,1048.36058
JFK,24.0,14.38475,3.623229,4.60312,12.65858,14.96014,16.398615,20.71404
LGA,24.0,14.96014,4.398433,4.60312,13.23397,14.96014,17.261700,23.01560


In [16]:
feb12_weather.groupby("origin")["wind_speed"].max()

origin
EWR    1048.36058
JFK      20.71404
LGA      23.01560
Name: wind_speed, dtype: float64

In [17]:
feb12_weather.groupby("origin")["wind_speed"].mean()

origin
EWR    56.38822
JFK    14.38475
LGA    14.96014
Name: wind_speed, dtype: float64

In [18]:
feb12_weather.groupby("origin")["wind_speed"].median()

origin
EWR    13.80936
JFK    14.96014
LGA    14.96014
Name: wind_speed, dtype: float64

### Answer

When I looked at the wind speeds, Newark (EWR) had a value of 1048.36 mph. This is clearly an outlier and not a realistic wind speed, so I did not use it when comparing the airports. After accounting for the outlier, **LaGuardia Airport (LGA)** had the highest valid wind speed at about 23.02 mph. Therefore, LGA had the windiest weather on February 12, 2013.